README.md
### This project wrriten in order to get alertmanager event and send webhooks acorrdinaly 

## prometheus/ thanos_rule rule
```
  - alert: example_alert_awx
    expr: job:request_latency_seconds:mean5m{job="myjob"} > 0.5
    for: 5m
    labels:
      severity: P3
    annotations:
      alertname: example_alert_awx
      summary: "example_alert_awx"
      description: "example_alert_awx"

```
## alertmanager config example:


```
receivers:
- name: webhooker_opsgenie
  opsgenie_configs:
  - api_key: OPSGENIE_API_KEY
    send_resolved: true
    priority: 'P3'
  webhook_configs:
  - url: 'http://webhookerurl.com:5000/postjson'
    send_resolved: true
```


![trigger](images/flow.png)


in our case it will triggers awx templates

### how to run
docker run --name=mysql-webhooker  --env=MYSQL_ROOT_PASSWORD=****  --network=host -p 0.0.0.0:3306:3306 --restart=on-failure --log-opt max-size=100m  --detach=true mysql:5.7 mysqld

docker build . -t your_docker_image

docker run  -e smtp_username=**** -e smtp_password=**** -e awx_username=***** -e awx_password=****  -e mysql_password=**** jenkins_token=****  your_docker_image

### how to change configuration

change config.yaml 

### Awx URL var
url: "https://awx-example-url.com/api/v2/job_templates/43/launch/"

refrence : 
https://docs.ansible.com/ansible-tower/3.2.6/html/towerapi/launch_jobtemplate.html

### Alertmanager alerts 
limit:
 ## alert name
 sin_mirror:  ''' {"limit": "sin-mirror-maker","extra_vars": {"command": "start","kafka_cluster_name": "kafka-etl-9"} '''

### UI with some info such as alert configuartion 

http://your_host:5000/



### how to test it

curl -vX POST http://127.0.0.1:5000/postjson  --header "Content-Type: application/json" -d @test.json


###json example
```
{
	"receiver": "webhooker_opsgenie",
	"status": "firing",
	"alerts": [{
		"status": "firing",
		"labels": {
			"alertname": "file_transfer",
			"instance": "was-example-sdp5.stapp.com:8004",
			"severity": "p2",
			"team": "sdp_autohealing"
		},
		"annotations": {
			"alertname": "file_transfer",
			"description": "there are files older than 70 minutes",
			"instance": "was-example-sdp5.stapp.com:8004",
			"summary": "file transfer alert"
		},
		"startsAt": "2020-03-19T08:31:33.027000593Z",
		"endsAt": "0001-01-01T00:00:00Z"

	}],
	"groupLabels": {
		"alertname": "file_transfer",
		"instance": "was-example-sdp5.stapp.com:8004"
	},
	"commonLabels": {
		"alertname": "file_transfer",
		"instance": "was-example-sdp5.stapp.com:8004",
		"severity": "p2",
		"team": "sdp_autohealing"
	},
	"commonAnnotations": {
		"alertname": "file_transfer",
		"description": "there are files older than 70 minutes",
		"instance": "was-example-sdp5.stapp.com:8004",
		"summary": "file transfer alert"
	}

}
```

## for debugging purpose 

http://webhooker-url.com:5000/

* at the top of the page you will see 500 last trigger alerts 

![trigger](images/last_trigger.png)

* at the buttom of the page you can see the last 20 json alerts who sent from alertmanager 


![trigger](images/last_json.png)


TODO:

* add docker-compose file
* add helm charts 

CHANGELOG.md
# CHANGELOG

<!--- next entry here -->

## 0.0.99
2021-09-02

### Fixes

- flink batch app auto fix (0c59fdc4bec0713009b7348e81bf0ea69d8333d9)

## 0.0.98
2021-08-19

### Fixes

- comment short circuit automation (b5d8f42bbd48c9d022e48e941878c0071b44b701)

## 0.0.97
2021-08-18

### Fixes

- blocked 200 (7d23432b4e4a41f3dacc6aa7175bae7fc6e96b7d)

## 0.0.96
2021-08-17

### Fixes

- catalog restore fix (b934fd79b4ede2b5a34f17659c498ba73a367b87)

## 0.0.95
2021-07-27

### Fixes

- short circuit block 100 (105ba5eacc3746d2746d16dd03795e46aee13e14)
- add support in multi instance (c2c253fa336e9f67438ddb2f87d0647fe60e6394)
- remove deubug (f9e9817df926bf44a47222c851a19cb390dfd41e)

## 0.0.94
2021-07-25

### Fixes

- make json more visible (c759f3fef0a12fe472e9f061ab553d452bef3ccd)

## 0.0.93
2021-07-22

### Fixes

- add auto restart role monolith add block (128c08e1596bfdbd6b67ffdd543a1fa4a6f2ed12)

## 0.0.92
2021-07-22

### Fixes

- add auto restart  monolith (9d91cf334d8563801c8f62bd73cc0f672d70a16b)

## 0.0.91
2021-07-18

### Fixes

- add README add debug (b57f905f847d54ff5e642ca8059de697d69cc3ab)

## 0.0.90
2021-07-15

### Fixes

- update config (b1433fd55d52e06c8225ac048693d165da3c0921)
- short_circuit_servers (af1d2ed6382ad2c1dc2e00237d94581b9a100dba)

## 0.0.89
2021-07-15

### Fixes

- sc instance (aa3814ade18c73afca632d2a6779cfb43c455259)

## 0.0.88
2021-07-15

### Fixes

- short_circuit1 (40a68044a7d61a8856eae5cb531c0dfa118314cd)

## 0.0.87
2021-07-14

### Fixes

- add support in collection (7c77f75bcefc5fb0580c03105e0b067c587474d2)
- add support in last 500 alerts in ui (5274c739db40cbe6d6025332e2af7a8e2adb93f0)

## 0.0.86
2021-07-07

### Fixes

- auto restore catalog sync (d1f6d6dee7ae4f797fff002920ed4444cbd7c887)

## 0.0.85
2021-06-15

### Fixes

- resume nobid from overshoot (92403db0dba935a50d4237981a61ab07fa267c65)

## 0.0.84
2021-06-14

### Fixes

- temp disable nobid from overshoot (d63f913a6ca77bc2748668a7cc14fb287d4f7448)

## 0.0.83
2021-05-25

### Fixes

- resloved (3887908464ebd48492e666e539bf54c5143e96da)
- nonRtadvent (8338d2705be554c33da672af74a5f52ec450795e)

## 0.0.82
2021-05-25

### Fixes

- test (b5f7d09457c86994e9ce743c44cd86a9c53636b6)
- add nonRtAdEvent retention (2e37ffdaaaf9f7129f07048669ed33e09b0447f3)
- nonRtAdEvent (a58db4122bceaecb3e949792739f8a39b9618740)
- duplicates (d1c98794d34d978c9df9c44e3b064229b9518746)

## 0.0.81
2021-05-25

### Fixes

- add nonRtAdEvent (99d66ac5ac2b494f61bfc95f797c64133e984dca)

## 0.0.80
2021-05-11

### Fixes

- eyal request temp disable file transfer auto heal (586cccb5f1c176276b23e824e32f4d6f7e04953c)

## 0.0.79
2021-03-18

### Fixes

- overshoot enable 5min block (cde933dd516fc9f9278d67443a77d6f6a73331d8)

## 0.0.78
2021-03-03

### Fixes

- flink audience restart for working hours (33aa3c203f55687afe928ad81266ad65379c6e33)

## 0.0.77
2021-02-21

### Fixes

- flink restart block duration 10 minutes (c323f6f6323734ccc0dd9e860db1691db2471554)

## 0.0.76
2021-02-16

### Fixes

- naming (a2af1b5854fb7a82ad037ed67e3f69a14cf01b4e)

## 0.0.75
2021-02-16

### Fixes

- nobid from overshooting (17bc28e9d2c2cc344ee89691597f83098ad561d4)

## 0.0.74
2021-02-02

### Fixes

- add retention resolved (57950342ff9026a0fb6b7fc85e46f48ed2705e33)
- add retention resolved (1e8ca7a07d78496989ceac37832c7a1a3cc2fdb6)

## 0.0.73
2021-02-02

### Fixes

- add alerts (64616e165fbeb3cf8b88f927aac98ac110f86597)

## 0.0.72
2021-01-24

### Fixes

- enable flink autorestart again (42135284bc406f29c8fa5349be98647344980f6c)

## 0.0.71
2021-01-24

### Fixes

- stop app autorestart (2932d5a9b787b2b538a291013435efb88388f4f6)

## 0.0.70
2021-01-17

### Fixes

- **config:** rtb nobid (87b05134d58d647404bc304d02890cbc2aa96263)
- roman webhook gitlab testing (df6b97354cd08b38e275847ec23aa413b0ea08a8)

## 0.0.69
2020-12-23

### Fixes

- **conf:** was mongodb lag (265ad545e611d2215c9ee327d716e35589a40669)

## 0.0.68
2020-12-02

### Fixes

- remove test (8ca59af3bd431afe13b18642f810cd048d5baad1)

## 0.0.67
2020-11-29

### Fixes

- alert name (8dde2f71575ef6ec615e2c8d8231484d8aabf9dd)
- chnage name (29853e87f5d54ecad943559e4833e1336692ae99)

## 0.0.66
2020-11-26

### Fixes

- webhook counteradevent nobid automation (75bdbcfabfdc84c764208bd4740d590c483c8929)

## 0.0.65
2020-11-24

### Fixes

- add test alert (4263a6cbfb1bf2268cf8e494a21f69e69ae78fb8)
- webhooker support flink monitoring audience site (a5db7f29287a0104c4c023ed676414a64b25af47)

## 0.0.64
2020-11-18

### Fixes

- add resolve config to code (7d45f0d933921fe1d9c53e1d5a703dec50b6a64f)
- add email specific support (b4342adacb0407f117625e90382788c0a6fceccd)
- add startappservices nobid (96c6a9474a80c0a37c7206db0bb304e60967d388)
- add description (0f63d0485156413b9e8d9182a32e868e477fe867)

## 0.0.63
2020-11-04

### Fixes

- change alertname print location (e773412f6d9a4e7207836557d1fe5ec12e7ce219)

## 0.0.62
2020-08-30

### Fixes

- add limit to select (fc031b8059b6f404ceca63b0e865e85b38326e31)

## 0.0.61
2020-08-20

### Fixes

- add blocked to ui (8107afd6b7c50e7c2c48a70c8744af5fde38a8b1)
- add order by time (136b97b38226d170819b24e1bfea597653c1a7c6)

## 0.0.60
2020-08-20

### Fixes

- pass (d982143ab3456f1b7690109a63b1cf5ea73042df)

## 0.0.59
2020-08-20

### Fixes

- change thresholds to adwin and adimpression (0f7b7fc579da1c72b661d24776164fad7ca0fcd5)

## 0.0.58
2020-08-20

### Fixes

- pasword issue (71f55ce68d765104d7d1c9c4b43a00b5b1a17b11)

## 0.0.57
2020-08-20

### Fixes

- fix pass issue (7aac22ea04446b9007425a8b3086407397c1e42b)

## 0.0.56
2020-08-20

### Fixes

- add mysql-connector-python (03964a3fe4e37dbdb3c4f53540ad67bd40f645e7)

## 0.0.55
2020-08-20

### Fixes

- add logger (ea9dd28f144f159e8889437aa2713fc4c9b71f2d)
- logging (a82dce01bd9e3a46a637d82fa475531925517735)

## 0.0.54
2020-08-19

### Fixes

- job (d9609b16097f199a812c60a999a5ec3255b4e0b0)
- add block option to alert (86c880566e11121a4667e1155637259ca8eed7be)

## 0.0.53
2020-08-03

### Fixes


select_mysql.py
# importing required library
import mysql.connector


class Mysql_query_class():
    def __init__(self,query,alert,password,mysql_host):
        self.query = query
        self.password = password
        self.alert = alert
        self.host = mysql_host
        self.dataBase = mysql.connector.connect(
            host=self.host,
            user="root",
            passwd=password,
            database="mysql")
        self.cursorObject = self.dataBase.cursor()

    def mysql_get_results(self):
        query = "select  * from alert order by time desc limit 2000"
        self.cursorObject.execute(query)
        # disconnecting from server
        myresult = self.cursorObject.fetchall()
        #print(myresult)
        self.dataBase.close()
        return myresult

    def mysql_query(self):
        query = self.query
        self.cursorObject.execute(query)
        # disconnecting from server
        myresult = self.cursorObject.fetchall()
        print(myresult)
        self.dataBase.close()
        if myresult:
            return True
        else:
            return False

    def mysql_insert(self):
        query = self.query
        self.cursorObject.execute(query)
        # disconnecting from server
        print('insert alert into mysql: ' +  query)
        self.dataBase.commit()
        self.dataBase.close()

    def mysql_drop_create_table(self):
        #self.query = "CREATE TABLE IF NOT EXISTS mysql.alert ( name  VARCHAR(255), time TIMESTAMP );"
        query = self.query
        self.cursorObject.execute(query)
        # disconnecting from server
        #self.dataBase.commit()
        self.dataBase.close()




#query = "select  * from alert ;"

#password = "test"

#alert = 'test'

#Mysql_query_obj = Mysql_query_class(query,alert,password)
#test = Mysql_query_obj.mysql_get_results()

#print(test)

#for i in test:
#    print(i)


### insert
#query = "INSERT INTO alert (name) values ('test');"
#password = "test"
#alert = 'test'
#Mysql_query_obj = Mysql_query_class(query,alert,password)
#Mysql_query_obj.mysql_insert()

jenkins_class.py
import jenkins


# It comes with pip module python-jenkins
# use pip to install python-jenkins

# Jenkins Authentication URL



class Jenkins_class():
    def __init__(self ,JENKINS_URL,JENKINS_USERNAME,JENKINS_TOKEN,JOB):
        self.JENKINS_URL = JENKINS_URL
        self.JENKINS_USERNAME = JENKINS_USERNAME
        self.JENKINS_TOKEN = JENKINS_TOKEN
        self.JENKINS_JOB = JOB
        self.jenkins_server = jenkins.Jenkins(self.JENKINS_URL, username=JENKINS_USERNAME, password=JENKINS_TOKEN)
        user = self.jenkins_server.get_whoami()
        version = self.jenkins_server.get_version()
        #print ("Jenkins Version: {}".format(version))
        print ("Jenkins User: {}".format(user['id']))

    def build_job(self):
        next_build_number = self.jenkins_server.get_job_info(self.JENKINS_JOB)['nextBuildNumber']
        self.jenkins_server.build_job(self.JENKINS_JOB, token=self.JENKINS_TOKEN)
        print(next_build_number)
        #time.sleep(10)
        #build_info = self.jenkins_server.get_build_info(self.JENKINS_JOB, next_build_number)
        #return build_info


email_notification.py
import smtplib
from email.mime.multipart import MIMEMultipart
from email.mime.text import MIMEText
from email.mime.base import MIMEBase
from email import encoders
# coding: utf-8


alert = 'sin_mirror'
toaddr= 'avishayafik@gmail.com'
subject='restart was done ' + alert



class email():
    def __init__(self,toaddr,subject,fromaddr,smtp,smtp_port,username,password,email_message):
        #self.alert = alert
        self.toaddr = toaddr
        self.subject = subject
        self.fromaddr = fromaddr
        self.smtp = smtp
        self.smtp_port = smtp_port
        self.username = username
        self.password = password
        self.email_message = email_message
        #self.email_message = email_message

    def send_email(self):
        space = " "
        msg = MIMEMultipart()
        msg['From'] = self.fromaddr
        msg['To'] = self.toaddr
        msg['Subject'] = self.subject
        body = ""
        p = MIMEBase('application', 'octet-stream')
        # creates SMTP session
        try:
            s = smtplib.SMTP(self.smtp, self.smtp_port)
            s.starttls()
            s.login(self.username, self.password)
            # Converts the Multipart msg into a string
            text = msg.as_string()
            # sending the mail
            s.sendmail(self.fromaddr, self.toaddr.split(','),text)
            # print "sent email successfully"
        except Exception:
            pass

    def send_email_html(self,color):
        space = " "
        msg = MIMEMultipart()
        msg['From'] = self.fromaddr
        msg['To'] = self.toaddr
        msg['Subject'] = self.subject
        body = ""
        p = MIMEBase('application', 'octet-stream')
        if color == "green":
            html = """\
            <!DOCTYPE html>
            <html>
               <body style="background-color:green;">
                  <p>    </p>
                  <p></p>
                  <p></p>
                  <p></p>
                  <p></p>
                  <p></p>
                  <h1>"%s"</h1>
                  <p></p>
                  <p></p>
                  <p></p>
                  <p></p>
                  <p></p>
                  <p>      </p>
               </body>
            </html>
            """ % (self.email_message)
        else:
            html = """\
            <!DOCTYPE html>
            <html>
               <body style="background-color:red;">
                  <p>    </p>
                  <p></p>
                  <p></p>
                  <p></p>
                  <p></p>
                  <p></p>
                  <h1>"%s"</h1>
                  <p></p>
                  <p></p>
                  <p></p>
                  <p></p>
                  <p></p>
                  <p>     </p>
               </body>
            </html>
            """ % (self.email_message)
        HTML_BODY = MIMEText(html, "html")
        msg.attach(HTML_BODY)
        # creates SMTP session
    #try:
        s = smtplib.SMTP(self.smtp, self.smtp_port)
        s.starttls()
        s.login(self.username, self.password)
        # Converts the Multipart msg into a string
        message = msg.as_string(msg)
        # sending the mail
        s.sendmail(self.fromaddr, self.toaddr.split(','),message)
                # print "sent email successfully"
     #   except Exception:
     #       pass


web-hooker.py
from flask import Flask
app = Flask(__name__)
from flask import Flask
from flask import request
app = Flask(__name__)
import requests
from jenkins_class import Jenkins_class
from argo_events import argo_events
import yaml
import os
from email_notification import email
from flask import Flask, render_template
#from uptime import uptime
import time
from prometheus_flask_exporter import PrometheusMetrics
from select_mysql import Mysql_query_class
import logging, sys
app = Flask(__name__)
metrics = PrometheusMetrics(app)
import json


startTime = time.time()

JSON_DISPLAY = ""
JSON_DISPLAY_LIST = []

logger = logging.getLogger('webhooker' + ' ')
logger.setLevel(logging.INFO)
ch = logging.StreamHandler(sys.__stdout__)
ch.setLevel(logging.INFO)
formatter = logging.Formatter('%(asctime)s - %(lineno)s - %(levelname)s - %(message)s')
ch.setFormatter(formatter)
logger.addHandler(ch)


data = yaml.load(open('config.yaml'))
print ("alert config:")
print(data['limit'])

jenkins_token = (os.environ['jenkins_token'])
username = (os.environ['awx_username'])
password = (os.environ['awx_password'])
smtp_username = (os.environ['smtp_username'])
smtp_password = (os.environ['smtp_password'])
mysql_password = (os.environ['mysql_password'])
mysql_host = (data['mysql_host'])


class awx():
    def __init__(self,url,data,username,password):
        self.data = data
        self.url = url
        self.username = username
        self.password = password
    def awx(self):
          timestamp = time.strftime("%Y-%m-%d %X")
          response = requests.post(self.url, auth=(self.username, self.password),data=self.data, headers={"content-type": "application/json"},timeout=3)
          print  (timestamp + ': running http hook , see the below response :')
          print  (response.content)

class flask_monitor():
    def __init__(self,path):
        self.path = path



    def run_prom(self):
          response = requests.get('http://127.0.0.1:5000/alerts/' + self.path)
          print  ('send http request to /alerts :')
          print(response.content)



@app.route('/postjson', methods=['POST'])
def post():
    timestamp = time.strftime("%Y-%m-%d %X")
    app.logger.info('Info')
    logger.info('request json: '+  str(request.is_json))
    content = request.get_json()
    global JSON_DISPLAY_LIST
    global JSON_DISPLAY
    JSON = request.get_json()
    JSON_DISPLAY = JSON
    JSON_DISPLAY = json.dumps(JSON_DISPLAY)
    JSON_DISPLAY_LIST.append(JSON_DISPLAY)
    #JSON_DISPLAY = JSON_DISPLAY.replace("'", '"')
    alerts_list  = content['alerts']
    status = content['alerts'][0]['status']
    content = (content['commonLabels']['alertname'])
    #logger.info('alertname:', content)
    try:
       instance = (JSON['commonLabels']['instance'])
       #print(instance)
    except Exception:
       pass
    ### add flink app support
    try:
       flink_app = (JSON['commonLabels']['metric_name'])
       dc = (JSON['commonLabels']['site'])
       collection = (JSON['commonLabels']['collection'])
    except Exception:
       pass
    print_action = False
    blocked_alert = False
    for key in data['limit'].items():
        if content in (key[1]['alertname']) and status == (key[1]['status']):
               json_to_send = key[1]['alertname'][content]
               try:
                   ##### support in multi instnace limit
                   instance_list = []
                   for i in alerts_list:
                       instance = i['labels']['instance']
                       instance = instance = instance.split(':')
                       instance = instance[0]
                       # print(instance)
                       instance_list.append(instance)
                   instance_list = str(instance_list)
                   instance_list = instance_list.replace("[", "")
                   instance_list = instance_list.replace("'", "")
                   instance_list = instance_list.replace("]", "")
                   #instance = instance.split(':')
                   #instance =  instance[0]
                   json_to_send = json_to_send.replace('INSTANCE',instance_list)
                   logger.info(timestamp + ': vars were replaced to '+ json_to_send)
               except Exception:
                   pass
               try:
                   json_to_send = json_to_send.replace('FLINK_APP', flink_app)
                   json_to_send = json_to_send.replace('DC', dc)
                   json_to_send = json_to_send.replace('COLLECTION', collection)
               except Exception:
                   pass
               ### check in mysql if it is blocked
               try:
                   query = "select  * from alert WHERE time > DATE_SUB(NOW(), INTERVAL '%s' MINUTE)  and name='%s' and vars='%s';" % ((key[1]['blocked']),content,json_to_send)
                   Mysql_query_obj = Mysql_query_class(query, alert, mysql_password,mysql_host)
                   blocked_alert = Mysql_query_obj.mysql_query()
               except Exception:
                   pass
               if blocked_alert == False:
                   logger.info(timestamp + ': send http webhook to:')
                   logger.info(timestamp + ': ' + (key[1]['url']))
                   logger.info(timestamp + ': vars sent:' + json_to_send )
                   #### send  alert awx, argo or jenkins
                   if key[1]['type'] == 'argo_workflow':
                        argo = argo_events((key[1]['url']),json_to_send,username,password)
                        argo.argo_run()
                   elif key[1]['type'] == 'jenkins':
                       run_jenkins = Jenkins_class((key[1]['url']),json_to_send,jenkins_token,key[1]['alertname']['job'])
                       run_jenkins.build_job()
                   else:
                       runawx = awx((key[1]['url']), json_to_send, username, password)
                       runawx.awx()

                   ### send email
                   logger.info('sending email to: '+ key[1]['email'] )
                   try:
                       if key[1]['status'] == 'resolved':
                           color = 'green'
                       else:
                           color = 'red'
                       email_meassge = key[1]['email_message']
                       subject = key[1]['email_message']
                       Email = email(fromaddr="no-noreply@startapp.com", toaddr=key[1]['email'],smtp=data['smtp'], smtp_port=data['smtp_port'],
                                     subject=subject, email_message=email_meassge, username=smtp_username, password=smtp_password)
                       Email.send_email_html(color)
                   except Exception:
                       pass
                       email_meassge = ""
                       subject = content + ' alert was running awx job'
                       Email = email(fromaddr="no-noreply@startapp.com", toaddr=key[1]['email'],smtp=data['smtp'], smtp_port=data['smtp_port'],
                                     subject=subject, email_message=email_meassge, username=smtp_username, password=smtp_password)
                       Email.send_email()
                   print_action = True
                   ### send http request to register it prometheus exporter
                   run_flask_monitor = flask_monitor(content)
                   run_flask_monitor.run_prom()
                   if print_action :
                       logger.info(timestamp + ': task was running for  ' + content)
                       ### insert into mysql so it will be blocked next time
                       query = "INSERT INTO alert (name,vars) values ('%s','%s');" % (content,json_to_send)
                       Mysql_query_obj = Mysql_query_class(query, alert, mysql_password,mysql_host)
                       blocked_alert = Mysql_query_obj.mysql_insert()
                   else:
                       logger.info (timestamp + ': Nothing to do')
                   logger.info(content)
               else:
                   logger.warning(content + " : is blocked for '%s' minutes , see config.yaml" %str((key[1]['blocked'])))
    return content

@app.route("/")

def home():
    run_result_obj = Mysql_query_class(query, alert, mysql_password,mysql_host)
    run_results = run_result_obj.mysql_get_results()
    global  JSON_DISPLAY_LIST
    JSON_DISPLAY_LIST =  JSON_DISPLAY_LIST[-20:]
    return render_template("home.html",rules=data['limit'],uptime=time.time() - startTime,run_results=run_results,json_results=JSON_DISPLAY_LIST)

## making sure there is alert table
query = "CREATE TABLE IF NOT EXISTS mysql.alert ( name  VARCHAR(255), vars VARCHAR(255), time TIMESTAMP );"
alert = 'test'
Mysql_query_obj = Mysql_query_class(query, alert, mysql_password,mysql_host)
Mysql_query_obj.mysql_drop_create_table()

#run_results = Mysql_query_obj.mysql_get_results()


## running flask webservice
app.run(host='0.0.0.0', port=5000)
